rotkehlchen/globaldb/handler.py

```diff
@@ -839,46 +862,176 @@ def get_evm_tokens(
         - Protocol for which to return tokens
         - ignore_spam (default True) to filter out "spam protocol" assets
         """
+        return GlobalDBHandler._get_tokens_from_db(
+            table_name='evm_tokens',
+            token_deserializer=EvmToken.deserialize_from_db,
+            token_type_name='EvmToken',
+            exceptions=exceptions,
+            protocol=protocol,
+            ignore_spam=ignore_spam,
+            chain_filter=('chain', chain_id.serialize_for_db()),
+            needs_underlying_tokens=True,
+            extra_columns=['chain'],
+        )
+
+    @staticmethod
+    def _get_single_token_from_db(
+            table_name: str,
+            token_deserializer: Callable,
+            token_type_name: str,
```

Comment:
instead of those two classes can't you just pass `EvmToken` and `SolanaToken`? You have the name and then you can call klass.deserialze()